### PR TITLE
Devtooling-198: Refactored user_roles resource

### PR DESCRIPTION
**Pull request by BrianMoyles**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -161,6 +162,7 @@ func registerResources() {
 	lineBaseSettings.SetRegistrar(regInstance)              //Registering Line Base Settings
 	edgesTrunk.SetRegistrar(regInstance)                    //Registering Edges Trunk Settings
 	resourceExporter.SetRegisterExporter(resourceExporters) //Registering register exporters
+	user_roles.SetRegistrar(regInstance)                    //Registering user roles
```
Comment (dginty4) [inline main.go:165]: should be camelCase

File: genesyscloud/user_roles/resource_genesyscloud_user_roles.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+func createUserRoles(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	userID := d.Get("user_id").(string)
+	d.SetId(userID)
+	log.Printf("Creating roles for user %s", d.Id())
+	return updateUserRoles(ctx, d, meta)
+}
+
+func readUserRoles(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*genesyscloud.ProviderMeta).ClientConfig
+	authAPI := platformclientv2.NewAuthorizationApiWithConfig(sdkConfig)
```
Comment (carnellj-genesys) [inline genesyscloud/user_roles/resource_genesyscloud_user_roles.go:32]: Need to extract APi calls out into a Proxy

**Review by carnellj-genesys — CHANGES_REQUESTED** [review]

File: genesyscloud/user_roles/genesyscloud_user_roles_init_test.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+	providerResources = make(map[string]*schema.Resource)
+	providerDataSources = make(map[string]*schema.Resource)
+
+	regInstance := &registerTestInstance{}
+
+	regInstance.registerTestResources()
+	regInstance.registerTestDataSources()
+}
+
+// TestMain is a "setup" function called by the testing framework when run the test
```
Comment (HemanthDogiparthi12) [inline genesyscloud/user_roles/genesyscloud_user_roles_init_test.go:57]: when the test is run

File: genesyscloud/user_group_role_common.go [hunk 4]
```diff
@@ -16,7 +16,7 @@ import (
 )
 
 var (
-	roleAssignmentResource = &schema.Resource{
+	RoleAssignmentResource = &schema.Resource{
```
Comment (HemanthDogiparthi12) [inline genesyscloud/user_group_role_common.go:19]: move this file as well to its own package. most likely this wil be used by users and groups. and add a proxy for this.

**Review by carnellj-genesys — APPROVED** [review]

**Pull request merged**